### ISSUE TITLE
[bug]: Tooltip in Sheet shows tooltip when sheet is opened

### ISSUE THREAD

**Issue by Weldawadyathink** [shadcn-ui/ui#3439]
### Describe the bug

Insert a Tooltip component into a sheet component. When the sheet is opened, the tooltip will popup instantly without hovering. Often it will jump around the screen as the sheet animates. 

### Affected component/components

Sheet, Tooltip

### How to reproduce

```
export default function App() {
  return (
    <>
      <Sheet>
        <SheetTrigger>Open</SheetTrigger>
        <SheetContent>
          <SheetHeader>
            <SheetTitle>Are you absolutely sure?</SheetTitle>
            <SheetDescription>
              This action cannot be undone. This will permanently delete your
              account and remove your data from our servers.
            </SheetDescription>
          </SheetHeader>

          <TooltipProvider>
            <Tooltip>
              <TooltipTrigger>Hover</TooltipTrigger>
              <TooltipContent>
                <p>Add to library</p>
              </TooltipContent>
            </Tooltip>
          </TooltipProvider>
        </SheetContent>
      </Sheet>
    </>
  );
}
```

Click the Open trigger. Notice that the "Add to library" tooltip is shown on screen before hovering over the tooltip trigger. 

### Codesandbox/StackBlitz link

[CodeSandbox Replication](https://codesandbox.io/p/devbox/tooltip-in-sheet-bug-x94v3l?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clusk5m2400073b6f8kp8rmot%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clusk5m2400023b6fz42jiju8%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clusk5m2400043b6f422s1w6k%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clusk5m2400063b6flehipl3j%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B60.27662214436903%252C39.72337785563097%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clusk5m2400023b6fz42jiju8%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clusk5m2400013b6f3zi8aib0%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252FREADME.md%2522%257D%255D%252C%2522id%2522%253A%2522clusk5m2400023b6fz42jiju8%2522%252C%2522activeTabId%2522%253A%2522clusk5m2400013b6f3zi8aib0%2522%257D%252C%2522clusk5m2400063b6flehipl3j%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clusk5m2400053b6f80o1kghz%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_PORT%2522%252C%2522taskId%2522%253A%2522dev%2522%252C%2522port%2522%253A5173%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clusk5m2400063b6flehipl3j%2522%252C%2522activeTabId%2522%253A%2522clusk5m2400053b6f80o1kghz%2522%257D%252C%2522clusk5m2400043b6f422s1w6k%2522%253A%257B%2522id%2522%253A%2522clusk5m2400043b6f422s1w6k%2522%252C%2522activeTabId%2522%253A%2522clusk640m003s3b6f9izsci3u%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clusk5m2400033b6fhgoi91h5%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522dev%2522%257D%252C%257B%2522id%2522%253A%2522clusk640m003s3b6f9izsci3u%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clusk642y00e4dhipc9mg6tid%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

### Logs

```bash
No log messages are produced
```

### System Info

```bash
Can be replicated in Chromium-based Arc, Firefox, Chrome, and Safari. Testing was done on MacOS, but issue has appeared on Windows (Firefox and Chrome) and iOS (Safari).
```

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues

**Comment by hsolrac** [comment]
hi @Weldawadyathink, 

I was simulating this error here and I noticed something. This happens because in the spreadsheet, for some reason, it puts the focus on the first element of the Content button (Sheet.Content) which triggers the tooltip

I'm still trying to figure out how to fix this.

![image](https://github.com/shadcn-ui/ui/assets/50886743/2ba7938d-a5c7-4530-a92f-d0b13d975a93)
![image](https://github.com/shadcn-ui/ui/assets/50886743/8c4c7fae-a489-4120-94f4-0bcc2bba49c9)

**Comment by Weldawadyathink** [comment]
Hi @hsolrac,

That is incredibly helpful! With that info, it sounds like this might be intended behavior. If it is intended, a prop to disable that behavior would be nice to have. Either way, with your info, I figured out a workaround. Its pretty clunky, but functional. If you add a blank <button /> as the first element, the sheet will focus that instead of your actual controls. It doesn't seem to show up visually. I updated my codesandbox.io project with a sample of that workaround.

**Comment by hsolrac** [comment]
Yes, that's an alternative. I'm glad you got a "solution" for now.

**Comment by OmarAljoundi** [comment]
Hi [Weldawadyathink](https://github.com/Weldawadyathink)

There is a better way to solve this, you can use the prop provide by [RadixUI](https://www.radix-ui.com/primitives/docs/components/dialog) to **prevent auto focus** on open which is `onOpenAutoFocus`.

In your Sheet or Dialog content add this prop ` <SheetContent onOpenAutoFocus={(event) => event.preventDefault()}> `
thus, it will prevent the auto focusing when the Sheet or Dialog is opened.

**Comment by issam-seghir** [comment]
I'm using tabIndex to prevent focusing on the tooltip that opens when I launch the dialog, so instead, the focus is on the input field.

```ts
 <TooltipTrigger asChild >
    <Button
       type="button"
       variant="ghost"
       size="icon"
       className="size-6"
        tabIndex={-1} // Prevent it from receiving focus on dialog open
    >
              <HelpCircle className="size-4" />
              <span className="sr-only">
                      Help
               </span>
     </Button>
 </TooltipTrigger>
```

**Comment by shadcn** [comment]
This issue has been automatically closed due to one year of inactivity. If you’re still experiencing a similar problem or have additional details to share, please open a new issue following our current issue template. Your updated report helps us investigate and address concerns more efficiently. Thank you for your understanding! (This is an automated message)
